test/JDBC/expected/temp_tables_oid-vu-verify.out

```diff
@@ -0,0 +1,121 @@
+-- Tables typically consume three OIDs - we checked this in permanent tables.
+create table #temp_tables_oid_t1(a int)
+create table #temp_tables_oid_t2(a int)
+select object_id('#temp_tables_oid_t2') - object_id('#temp_tables_oid_t1')
+go
+~~START~~
+int
+3
+~~END~~
+
+
+drop table #temp_tables_oid_t1
+drop table #temp_tables_oid_t2
+go
+
+create table temp_tables_oid_perm_tab_3(a int)
+go
+
+-- Should be empty here
+select relname from sys.babelfish_get_enr_list()
+GO
+~~START~~
+text
+~~END~~
+
+
+-- 1 entry expected, just the table. 
+CREATE TABLE #temp_tables_oid_t1(a int, b int)
+GO
+
+select count(*) from sys.babelfish_get_enr_list()
+GO
+~~START~~
+int
+1
+~~END~~
+
+
+-- 2 entries expected, the table and the index.
+CREATE INDEX temp_tables_oid_idx1 on #temp_tables_oid_t1(a)
+GO
+
+select count(*) from sys.babelfish_get_enr_list()
+GO
+~~START~~
+int
+2
+~~END~~
+
+
+-- 3 entries expected, the table and 2 index
+CREATE INDEX #temp_tables_oid_idx2 on #temp_tables_oid_t1(b)
+GO
+
+select count(*) from sys.babelfish_get_enr_list()
+GO
+~~START~~
+int
+3
+~~END~~
+
+
+drop table #temp_tables_oid_t1
+GO
+
+-- 0 entries expected
+select count(*) from sys.babelfish_get_enr_list()
+GO
+~~START~~
+int
+0
+~~END~~
+
+
+
+-- identity and primary key should be in ENR as well
+create table #temp_tables_oid_t1(a int identity primary key, b int)
```

Comment:
1/ Is there a way to test what happens if we can assign temp oid for the identity and table but we fail to allocate temp oid for pkey because no oid available?

Reply:
so, this is an interesting case i ran into while testing. i believe this might be the same as PG's default behavior with OID wraparound. 

here's what I believe happens, say if the buffer size is 4 for simplicity. 

1. I do `create table #t1...`, which consumes the first 3 oids. (call it OID 1000, 1001, 1002). 
2. I do `create table #t2...`. The table OID is the 4th oid (1004). 
3. Then, the buffer wraps around when we try to create the array type OID. In `GetNewOidWithIndex` and therefore `GetNewTempOidWithIndex`, PG calls `systable_beginscan` on `pg_type` with OID (1000), because, it wants to see if there is a collision. But it won't pickup OID 1000 because that's an entry in pg_class. So it says oh, looks like 1000 is free. And allows it to continue.

Reply:
It's hard to find a verified source because OID wraparound seems quite rare, but the code implies that they do, since they're only searching the same catalog for collisions in the OID gen code.

Reply:
PG documentation specifies: 

```
The oid type is currently implemented as an unsigned four-byte integer. Therefore, it is not large enough to provide database-wide uniqueness in large databases, or even in large individual tables.
```

and in the comments of GetNewObjectId: 

```
 * OIDs are generated by a cluster-wide counter.  Since they are only 32 bits
 * wide, counter wraparound will occur eventually, and therefore it is unwise
 * to assume they are unique unless precautions are taken to make them so.
```